src/codeu/chat/server/Server.java

```diff
@@ -15,295 +15,424 @@
+                timeline.scheduleNow(createSendToRelayEvent(
+                        author,
+                        conversation,
+                        message.id));
+            }
+        });
 
-    // New User - A client wants to add a new user to the back end.
-    this.commands.put(NetworkCode.NEW_USER_REQUEST,  new Command() {
-      @Override
-      public void onMessage(InputStream in, OutputStream out) throws IOException {
+        // New User - A client wants to add a new user to the back end.
+        this.commands.put(NetworkCode.NEW_USER_REQUEST, new Command() {
+            @Override
+            public void onMessage(InputStream in, OutputStream out) throws IOException {
 
-        final String name = Serializers.STRING.read(in);
-        final User user = controller.newUser(name);
+                final String name = Serializers.STRING.read(in);
+                final User user = controller.newUser(name);
 
-        Serializers.INTEGER.write(out, NetworkCode.NEW_USER_RESPONSE);
-        Serializers.nullable(User.SERIALIZER).write(out, user);
-      }
-    });
+                Serializers.INTEGER.write(out, NetworkCode.NEW_USER_RESPONSE);
+                Serializers.nullable(User.SERIALIZER).write(out, user);
+                logQueue.transactions.add("ADD-USER " + user.id.toString() + " " + user.name + " " + user.creation.inMs());
+            }
+        });
 
-    // New Conversation - A client wants to add a new conversation to the back end.
-    this.commands.put(NetworkCode.NEW_CONVERSATION_REQUEST,  new Command() {
-      @Override
-      public void onMessage(InputStream in, OutputStream out) throws IOException {
+        // New Conversation - A client wants to add a new conversation to the back end.
+        this.commands.put(NetworkCode.NEW_CONVERSATION_REQUEST, new Command() {
+            @Override
+            public void onMessage(InputStream in, OutputStream out) throws IOException {
 
-        final String title = Serializers.STRING.read(in);
-        final Uuid owner = Uuid.SERIALIZER.read(in);
-        final ConversationHeader conversation = controller.newConversation(title, owner);
+                final String title = Serializers.STRING.read(in);
+                final Uuid owner = Uuid.SERIALIZER.read(in);
+                final ConversationHeader conversation = controller.newConversation(title, owner);
 
-        Serializers.INTEGER.write(out, NetworkCode.NEW_CONVERSATION_RESPONSE);
-        Serializers.nullable(ConversationHeader.SERIALIZER).write(out, conversation);
-      }
-    });
+                Serializers.INTEGER.write(out, NetworkCode.NEW_CONVERSATION_RESPONSE);
+                Serializers.nullable(ConversationHeader.SERIALIZER).write(out, conversation);
+                logQueue.transactions.add("ADD-CONVERSATION " + conversation.id.toString() + " " + owner.toString() + " " +
+                        title + " " + conversation.creation.inMs());
```

Comment:
You'll want to print quotes around `title` here in case it contains whitespace (https://github.com/lysanderp41/CodeUSummer2017/issues/45).